synapse/rest/client/push_rule.py

```diff
@@ -53,26 +54,32 @@ def __init__(self, hs: "HomeServer"):
         self.notifier = hs.get_notifier()
         self._is_worker = hs.config.worker.worker_app is not None
         self._push_rules_handler = hs.get_push_rules_handler()
+        self._push_rule_linearizer = Linearizer(name="push_rules")
```

Comment:
I'm not sure if it matters, but this won't fix the issue if there are multiple workers involved. Would it make more sense to lock the user's push rule rows?

Reply:
Yeah that's a good point, I'm basing the change entirley on the current state where push rules can only be modified on the main process, this only works while that limitation remains (does that block merging as-is, not sure?).

Reply:
I had assumed we could move these to workers, my fault for not checking!